src/rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -177,18 +179,41 @@ function  azure_rdma_preparation()
 azure_rdma_preparation
 {{/ azRDMA }}
 {{/ reqAzRDMA }}
-# Write env to system-wide environment
-env | grep -E "^PAI|PATH|PREFIX|JAVA|HADOOP|NVIDIA|CUDA" > /etc/environment
+
+function write_user_command()
+{
+  local scriptPath="$1"
+
+  # Read an escaped string to an array and unescape
+  local lines=$'{{ taskData.command }}'
+  readarray -n 50 -t lines <<< "${lines}"
+  read -n 1000 -a lines <<< "${lines[@]}"
```

Comment:
`read` uses space as delimiter? it will split `lines` by space in my test.

Reply:
Yes, it does, but all whitespaces should come escaped at this point. Only unescaped should be from `\n`s. What `{{ }}` mustache braces do is a substitution of html-escaped string, which is not the same as bash-escaped. I think it's better just to escape all the characters in the command, substitute it with `{{{ }}}`, and then use `read`.

Reply:
the escaped characters have been reconfigured [here](https://github.com/Microsoft/pai/blob/master/src/rest-server/src/config/index.js#L32-L48), so `{{{ }}}` isn't needed.